api/controllers/DashboardController.js

```diff
@@ -87,6 +53,50 @@ var dashboardController = {
         });
     },
 
+    _calculateSortParams(req, res, userData, totalApplications) {
+        const pageSize = 20;
+        const currentPage = req.query.page || 1;
+        const offset = pageSize * (currentPage - 1);
+        const sortOrder = Number(req.query.sortOrder) || -1;
+        const isNumberPositive = Math.sign(sortOrder) === 1;
+        const direction = isNumberPositive ? 'asc' : 'desc';
+        const searchCriteria =
+            req.allParams().dashboardFilter || req.query.searchText || '';
+        //If user has specifically selected to filter by date (sortOrder eq 1 or -1)
+        //then we don't want to secondary sort by date again! But if a user filters by
+        //reference number for example, then secondary sort on the date as well.
+        const secondarySortOrder =
+            (sortOrder === 1 || sortOrder === -1) ? null : '1';
+        const secondaryDirection =
+            (sortOrder === 1 || sortOrder === -1) ? null : 'desc';
+        const storedProcedureArgs = {
+            replacements: {
+                userId: req.session.passport.user,
+                pageSize,
+                offset,
+                sortOrder: Math.abs(sortOrder).toString(),
+                direction,
+                queryString: '%' + searchCriteria + '%',
```

Comment:
should this be using `querystring.encode` or `encodeURIComponent()`? (These are both native node or js so no need to worry about another library), generally they are much safer and better tested than DIY-encoding

Reply:
I actually have no idea what the benefit of the percentages symbols are. I've done some research and they don't seem to be a wildcard operator for sequelize queries:
https://sequelize.org/master/manual/raw-queries.html

I think they mean if there are values that start or end with the searchCriteria they will be presented to the user.

Maybe I'm missing something but I don't know how useful the encodeURIComponent() function would be here.

Reply:
AH, my bad, this is a db queryString (not a URL one, so no, my suggestions wont help you). When you do %yourSearchString%, sql will try to search for that substring. If you just have one on either side, '%'+ searchCriteria, it will only search for when searchCriteria occurs at the end of the string. (and vice versa for searchCriteria + '%')